File: Leetcode/Easy Problems/876. Middle of the Linked List/dec-02-2023.cpp
/*
  December 02, 2023
  - Runtime Achieved: 3 ms 
  - Memory Achieved: 7.3 MB
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) { 
        ListNode* nodePace1 = head;
        ListNode* nodePace2 = head;
        while (nodePace1!=NULL && nodePace1->next!=NULL) {
            nodePace1 = nodePace1->next->next;
            nodePace2 = nodePace2->next;
        } 
        return nodePace2; 
    }
};

File: Leetcode/Easy Problems/876. Middle of the Linked List/dec-02-2023.js
/*
  December 02, 2023
  - Runtime Achieved: 50 ms
  - Memory Achieved: 41.7 MB
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    var stride = head;
    while(stride && stride.next) {
        head = head.next;
        stride = stride.next.next;
    }
    return head;
};
